File: HW9/huan1588/Unique/MallardDuck.cpp
#include <iostream>
#include "MallardDuck.h"
#include "Quack.h"
#include "FlyWithWings.h"
#include <memory>

MallardDuck::MallardDuck( ) {
   quackBehavior = std::make_unique<Quack>( );
   flyBehavior = std::make_unique<FlyWithWings>( );
}

void MallardDuck::display( ) {
   std::cout << "I’m a real mallard duck" << std::endl;
}

File: HW14/huan1588/MallardDuck.java
public class MallardDuck extends Duck {
   public MallardDuck( ) {
      quackBehavior = new Quack( );
      flyBehavior = new FlyWithWings( );
      eggBehavior = new LaysEggBroody();
   }
   
   public void display( ) {
      System.out.println("I’m a real mallard duck");
   }
}
